src/userWallet.ts

```diff
@@ -219,4 +222,61 @@ export abstract class UserWallet {
     await sendNotification({ user: this.user, title: `Your balance is $${balanceUsd} (${balanceSatsPrettified} sats)`, logger: this.logger })
   }
 
+  generate2fa = () => {
+    const { secret, uri, qr } = generateSecret({ name: yamlConfig.name, account: this.user.phone })
+    /*
+    { secret: 'XDQXYCP5AC6FA32FQXDGJSPBIDYNKK5W',
+      uri: 'otpauth://totp/My%20Awesome%20App:johndoe?secret=XDQXYCP5AC6FA32FQXDGJSPBIDYNKK5W&issuer=My%20Awesome%20App',
+      qr: 'https://chart.googleapis.com/chart?chs=166x166&chld=L|0&cht=qr&chl=otpauth://totp/My%20Awesome%20App:johndoe%3Fsecret=XDQXYCP5AC6FA32FQXDGJSPBIDYNKK5W%26issuer=My%20Awesome%20App'
+    }
+    */
+    return { secret, uri }
+  }
+
+  save2fa = async ({ secret, token }): Promise<boolean> => {
+    if(this.user.twoFactor.secret) {
+      throw new TwoFactorError("2FA is already set", {logger: this.logger})
+    }
+
+    const tokenIsValid = verifyToken(secret, token)
+
+    if (!tokenIsValid) {
+      throw new TwoFactorError(undefined, {logger: this.logger})
+    }
+
+    this.user.twoFactor.secret = secret
+
+    try {
+      await this.user.save()
+      return true
+    } catch (err) {
+      throw new DbError("Unable to save 2fa secret", {forwardToClient: true, logger: this.logger, level: 'error', err})
+    }
+  }
+
+  static validate2fa = ({ secret, token, logger }): boolean => {
```

Comment:
it's odds to have this one being a static and not the other one. why is that?

Reply:
I want to call this from inside `Lightning.ts`

Reply:
oh gotcha. but this is not a great design. some sure what the right solution here right now but something to think about.

Reply:
Added a comment to it. Idk what the right way to do this is. Maybe Samer can help here, but I am unable to tag him.